diff -up foomatic-filters-4.0.4/foomaticrip.c.format-string foomatic-filters-4.0.4/foomaticrip.c
--- foomatic-filters-4.0.4/foomaticrip.c.format-string	2013-06-14 15:51:27.929239770 +0100
+++ foomatic-filters-4.0.4/foomaticrip.c	2013-06-14 15:52:20.373442008 +0100
@@ -1600,7 +1600,7 @@ int main(int argc, char** argv)
         if (dontparse == 2) {
             /* Raw queue, simply pass the input into the postpipe (or to STDOUT
                when there is no postpipe) */
-            _log("Raw printing, executing \"cat %s\"\n\n");
+            _log("Raw printing, executing \"cat %s\"\n\n", postpipe->data);
             snprintf(tmp, 1024, "cat %s", postpipe->data);
             run_system_process("raw-printer", tmp);
             continue;
