diff -up foomatic-db-engine-4.0.3/configure.ac.perl foomatic-db-engine-4.0.3/configure.ac
--- foomatic-db-engine-4.0.3/configure.ac.perl	2009-12-04 09:26:14.236782616 +0000
+++ foomatic-db-engine-4.0.3/configure.ac	2009-12-04 09:29:08.819781206 +0000
@@ -184,13 +184,13 @@ else
   PREFIX=$prefix
 fi
 echo -n Checking whether Perl will find libraries installed under $PREFIX...
-if ( $PERL -V | egrep "^ *$PREFIX/lib(/perl[[0-9]]?|)(/site_perl|)/[[0-9]]" > /dev/null); then
+if ( $PERL -V | egrep "^ *$PREFIX/lib(/perl[[0-9]]?|)(/site_perl|)" > /dev/null); then
   PERLPREFIX=$PREFIX
   echo " yes"
 else
   echo " no"
   echo -n Checking whether Perl will find libraries installed under /usr/local...
-  if ( $PERL -V | egrep "^ */usr/local/lib(/perl[[0-9]]?|)(/site_perl|)/[[0-9]]" > /dev/null); then
+  if ( $PERL -V | egrep "^ */usr/local/lib(/perl[[0-9]]?|)(/site_perl|)" > /dev/null); then
     PERLPREFIX=/usr/local
     echo " yes"
     echo Installing the Perl libraries with prefix /usr/local.
diff -up foomatic-db-engine-4.0.3/configure.perl foomatic-db-engine-4.0.3/configure
--- foomatic-db-engine-4.0.3/configure.perl	2009-12-04 09:29:12.117906609 +0000
+++ foomatic-db-engine-4.0.3/configure	2009-12-04 09:29:34.709782105 +0000
@@ -4519,13 +4519,13 @@ else
   PREFIX=$prefix
 fi
 echo -n Checking whether Perl will find libraries installed under $PREFIX...
-if ( $PERL -V | egrep "^ *$PREFIX/lib(/perl[0-9]?|)(/site_perl|)/[0-9]" > /dev/null); then
+if ( $PERL -V | egrep "^ *$PREFIX/lib(/perl[0-9]?|)(/site_perl|)" > /dev/null); then
   PERLPREFIX=$PREFIX
   echo " yes"
 else
   echo " no"
   echo -n Checking whether Perl will find libraries installed under /usr/local...
-  if ( $PERL -V | egrep "^ */usr/local/lib(/perl[0-9]?|)(/site_perl|)/[0-9]" > /dev/null); then
+  if ( $PERL -V | egrep "^ */usr/local/lib(/perl[0-9]?|)(/site_perl|)" > /dev/null); then
     PERLPREFIX=/usr/local
     echo " yes"
     echo Installing the Perl libraries with prefix /usr/local.
diff -up foomatic-db-engine-4.0.3/Makefile.in.perl foomatic-db-engine-4.0.3/Makefile.in
