diff -up foomatic-filters-4.0.4/options.c.foomatic-rip-overrun foomatic-filters-4.0.4/options.c
--- foomatic-filters-4.0.4/options.c.foomatic-rip-overrun	2010-02-15 12:57:40.000000000 +0000
+++ foomatic-filters-4.0.4/options.c	2015-05-29 16:33:06.169987118 +0100
@@ -1087,8 +1087,9 @@ static void unhtmlify(char *dest, size_t
     const char *repl;
     struct tm *t = localtime(&job->time);
     char tmpstr[10];
+    size_t s;
 
-    while (*psrc && pdest - dest < size) {
+    while (*psrc && pdest - dest < size - 1) {
 
         if (*psrc == '&') {
             psrc++;
@@ -1153,8 +1154,12 @@ static void unhtmlify(char *dest, size_t
             }
 
             if (repl) {
-                strncpy(pdest, repl, size - (pdest - dest));
-                pdest += strlen(repl);
+	        s = size - (pdest - dest) - 1;
+                strncpy(pdest, repl, s);
+		if (s < strlen(repl))
+		  pdest += s;
+		else
+		  pdest += strlen(repl);
                 psrc = strchr(psrc, ';') +1;
             }
             else {
@@ -1600,8 +1605,8 @@ void read_ppd_file(const char *filename)
             /* "*FoomaticRIPOptionPrototype <option>: <code>"
                Used for numerical and string options only */
             opt = assure_option(name);
-            opt->proto = malloc(128);
-            unhtmlify(opt->proto, 128, value->data);
+            opt->proto = malloc(65536);
+            unhtmlify(opt->proto, 65536, value->data);
         }
         else if (!strcmp(key, "FoomaticRIPOptionRange")) {
             /* *FoomaticRIPOptionRange <option>: <min> <max>
